wgpu-hal/src/dx12/device.rs

```diff
@@ -976,50 +977,44 @@ impl crate::Device for super::Device {
                 });
                 bt.register += entry.count.map(NonZeroU32::get).unwrap_or(1);
             }
-            if ranges.len() > range_base {
-                let range = &ranges[range_base..];
-                parameters.push(Direct3D12::D3D12_ROOT_PARAMETER {
-                    ParameterType: Direct3D12::D3D12_ROOT_PARAMETER_TYPE_DESCRIPTOR_TABLE,
-                    Anonymous: Direct3D12::D3D12_ROOT_PARAMETER_0 {
-                        DescriptorTable: Direct3D12::D3D12_ROOT_DESCRIPTOR_TABLE {
-                            NumDescriptorRanges: range.len() as u32,
-                            pDescriptorRanges: range.as_ptr(),
+
+            let mut sampler_index_within_bind_group = 0;
+            for entry in bgl.entries.iter() {
+                if let wgt::BindingType::Sampler(_) = entry.ty {
+                    binding_map.insert(
+                        naga::ResourceBinding {
+                            group: index as u32,
+                            binding: entry.binding,
                         },
-                    },
-                    ShaderVisibility: conv::map_visibility(visibility_view_static),
-                });
-                info.tables |= super::TableTypes::SRV_CBV_UAV;
+                        hlsl::BindTarget {
+                            // Naga does not use the space field for samplers
+                            space: 255,
+                            register: sampler_index_within_bind_group,
```

Comment:
It's not great that this has a different meaning. We should either document this on the field or use an enum.

Reply:
Documented this on the field. I can't easily figure out in my head how to use an enum that works well.